src/main/java/frc/robot/hardware/motor/talonsrx/TalonSRXMotor.java

```diff
@@ -0,0 +1,81 @@
+package frc.robot.hardware.motor.talonsrx;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.TalonSRXControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.request.IRequest;
+import frc.robot.hardware.request.srx.AngleSRXRequest;
+import frc.robot.hardware.request.srx.DoubleSRXRequest;
+import frc.robot.hardware.signal.InputSignal;
+import frc.utils.Conversions;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+import org.littletonrobotics.junction.Logger;
+
+public class TalonSRXMotor implements ControllableMotor {
+
+	private final String logPath;
+	private final TalonSRX motor;
+	private final double gearRatio;
+
+	public TalonSRXMotor(String logPath, TalonSRX talonSRX, double gearRatio) {
+		this.logPath = logPath;
+		this.motor = talonSRX;
+		this.gearRatio = gearRatio;
+	}
+
+	@Override
+	public boolean isConnected() {
+		return true; // TODO
+	}
+
+	@Override
+	public void updateSignals(InputSignal... signals) {
+		for (InputSignal signal : signals) {
+			Logger.processInputs(logPath, signal);
+		}
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getSysidConfigInfo() {
+		return null;
+	}
+
+	@Override
+	public void resetPosition(Rotation2d position) {
+		motor.setSelectedSensorPosition(Conversions.angleToMagTicks(position, gearRatio));
+	}
+
+	@Override
+	public void setBrake(boolean brake) {
+		NeutralMode neutralMode = brake ? NeutralMode.Brake : NeutralMode.Coast;
+		motor.setNeutralMode(neutralMode);
+	}
+
+	@Override
+	public void stop() {
+		motor.set(ControlMode.PercentOutput, 0);
+	}
+
+	@Override
+	public void setPower(double power) {
+		motor.set(TalonSRXControlMode.PercentOutput, power);
+	}
+
```

Comment:
Delete

Reply:
Cannot delete, `void setPower(double power)` is part of `ControllableMotor` interface

Reply:
she is talking about the space

Reply:
got it